xpdan/callbacks_core.py

```diff
@@ -0,0 +1,218 @@
+"""module includes Callback classes for xpdacq/xpdan"""
+
+import os
+import datetime
+import numpy as np
+from bluesky.callbacks.core import CallbackBase
+import doct
+
+# supplementary functions
+def _timestampstr(timestamp):
+    """convert timestamp to strftime formate"""
+    timestring = datetime.datetime.fromtimestamp(float(timestamp)).strftime(
+        '%Y%m%d-%H%M%S')
+    return timestring
+
+
+class XpdAcqLiveTiffExporter(CallbackBase):
+
+    """Exporting tiff from given header(s).
+
+    It is a variation of bluesky.callback.broker.LiveTiffExporter class
+    It incorporate metadata and data from individual data points in
+    the filenames.
+
+    If also allow a room for dark subtraction if a valid dark_frame
+    metdata scheme is taken
+
+    Parameters
+    ----------
+    field : str
+        a data key, e.g., 'image'
+    data_dir_template : str
+        A templated for directory where images will be saved to.
+        It is expressed with curly brackets, which will be filled in with
+        the attributes of 'start', 'event', and (for image stacks) 'i',
+        a sequential number.
+        e.g., "/xpdUser/tiff_base/{start.sample_name}/"
+    data_fields : list, optional
+        a list of strings for data fields want to be included. default
+        is an empty list (not include any readback metadata in filename).
+    save_dark : bool, optionl
+        option to save dark frames, if True, subtracted images and dark
+        images would be saved. default is False.
+    dryrun : bool, optional
+        default to False; if True, do not write any files
+    overwrite : bool, optional
+        default to False, raising an OSError if file exists
+    db : Broker, optional
+        The databroker instance to use, if not provided use databroker
+        singleton
+    """
+
+    def __init__(self, field, data_dir_template,
+                 data_fields=[], save_dark=False,
+                 dryrun=False, overwrite=False, db=None):
+        try:
+            import tifffile
+        except ImportError:
```

Comment:
I would just let this break. The trace back may give us a bit more info than just the print statement. Also Tifffile is not the only import which could break this. Also also, I expect this to be a minor issue, we will require tifffile for the conda build and I think it is a dep for most of the NSLS-II stack.